lib/PreviewProvider.js

```diff
@@ -110,6 +110,17 @@ PreviewProvider.prototype = {
   },
 
   /**
+    * A decodeURIComponent that doesn't throw on invalid URI components
+    */
+  _safeDecodeURIComponent(str) {
+    try {
+      return decodeURIComponent(str);
+    } catch (e) {
+      return str;
```

Comment:
@oyiptong and I discussed this, let's verify that the URLs which come back are *not* URI encoded, and if that is the case, we shouldn't be decoding at all